### PR TITLE
Add documentation for Directus UI Components

### PR DESCRIPTION
**Pull request by estheragbaje**
This PR introduces a dedicated section in the extension documentation that focuses on how to use the Directus UI library within an extension project. Closes Issue directus/docs#307.

### CODE REVIEW

File: docs/.vitepress/data/sidebar.ts [hunk 1]
```diff
@@ -388,6 +388,10 @@ function sidebarDeveloperReference() {
 						},
 					],
 				},
+				{
+					link: '/extensions/leveraging-ui-components',
+					text: 'Leveraging UI Components',
```
Comment (phazonoverload) [inline docs/.vitepress/data/sidebar.ts:393]:
```suggestion
					text: 'Using Directus UI Components',
```

File: docs/extensions/leveraging-ui-components.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
+---
+description: Learn how to use the Directus UI components in your extension projects.
```
Comment (phazonoverload) [inline docs/extensions/leveraging-ui-components.md:2]:
```suggestion
description: Learn how to use the Directus UI components in your app extensions.
```

Removing 'projects' as that has a defined meaning within Directus

File: docs/extensions/leveraging-ui-components.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+contributors: Esther Agbaje
+---
+
+# Leveraging Directus UI Components
```
Comment (phazonoverload) [inline docs/extensions/leveraging-ui-components.md:6]:
```suggestion
# Using Directus UI Components
```

File: docs/extensions/leveraging-ui-components.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+Directus provides a set of reusable UI components that can be used in when building extensions. These components are
+designed to provide a consistent user experience across different projects and extensions.
```
Comment (phazonoverload) [inline docs/extensions/leveraging-ui-components.md:9]:
```suggestion
Directus provides a set of reusable UI components that can be used in when building extensions. These components are used by Directus itself, and are
designed to provide a consistent user experience across different parts of the Directus Data Studio.
```

File: docs/extensions/leveraging-ui-components.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+## Components Playground
+
+Explore the available components on the [Histoire Components Playground](https://components.directus.io/). Here, you can
+interact with different components and experiment with variations before implementing them in your project.
```
Comment (phazonoverload) [inline docs/extensions/leveraging-ui-components.md:14]:
```suggestion
Explore the available components in our [Directus Components Playground](https://components.directus.io/). Here, you can
see all available components and experiment with variations before implementing them in your extensions.
```

Comment (phazonoverload) [reply docs/extensions/leveraging-ui-components.md:14]: I don't think we need to expose the readers to this new term and tool - Histoire.

File: docs/extensions/leveraging-ui-components.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+![Directus UI Components](https://marketing.directus.app/assets/46d72f1a-5d9f-49f8-a09a-b2ffc5200812.gif)
```
Comment (phazonoverload) [inline docs/extensions/leveraging-ui-components.md:16]: This alt text needs to basically be a textual description of the important features of the playground shown in the gif.

File: docs/extensions/leveraging-ui-components.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+## Usage within Extensions
```
Comment (phazonoverload) [inline docs/extensions/leveraging-ui-components.md:18]:
```suggestion
## Extension Usage
```

File: docs/extensions/leveraging-ui-components.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+Directus UI components are globally registered, making them accessible throughout your extension project without the
+need for explicit imports. Here’s a basic example:
+
+```tsx
+<VButton>
+  My Button
+</VButton>
+```
```
Comment (phazonoverload) [inline docs/extensions/leveraging-ui-components.md:27]:
```suggestion
Directus UI components are globally registered, making them accessible throughout your extension project without the
need for explicit imports. Here’s an example:
```

Comment (phazonoverload) [reply docs/extensions/leveraging-ui-components.md:27]: A reminder: https://justsimply.dev/

File: docs/extensions/leveraging-ui-components.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+  My Button
+</VButton>
+
+// customize the style
+.my-button {
+  --v-button-color-hover: black;
+}
+```
+
+::: tip Explore CSS Variables
```
Comment (phazonoverload) [inline docs/extensions/leveraging-ui-components.md:47]:
```suggestion
::: tip Explore Component CSS Variables
```

File: docs/extensions/leveraging-ui-components.md [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+Here are some examples of CSS variables: `--border-normal`, `--foreground-normal` `-purple`, `--module-background`, and
+`--overlay-color`.
+
+::: tip Explore Light and Dark Theme CSS Variables
+
+Refer to our [codebase](https://github.com/directus/directus/tree/main/app/src/styles/themes) for a full list of CSS
+variables.
+
+:::
```
Comment (phazonoverload) [inline docs/extensions/leveraging-ui-components.md:70]:
```suggestion
The Directus UI components are designed with flexibility and customization in mind. However, you may need to create your own components using shared styling. Directus exposes several CSS variables for both light and dark themes.

Examples of CSS variables include `--border-normal`, `--foreground-normal` `-purple`, `--module-background`, and
`--overlay-color`.

::: tip Explore Light and Dark Theme CSS Variables

Refer to our [codebase](https://github.com/directus/directus/tree/main/app/src/styles/themes) for a full list of CSS
variables.

:::
```

**Review by phazonoverload — DISMISSED** [review]
Good, clear, helpful. Small suggested changes for clarity.  @paescuj can we get confirmation of when the components playground is brought up to date?

**Comment by phazonoverload** [comment]
There will also need to be a changeset for this PR

File: docs/extensions/leveraging-ui-components.md (hunk 8 above)
Comment (paescuj) [reply docs/extensions/leveraging-ui-components.md]: Was this supposed to remove the example?

**Review by paescuj — APPROVED** [review]

**Pull request merged**